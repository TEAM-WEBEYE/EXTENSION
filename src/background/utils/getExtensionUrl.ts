export function getExtensionUrl(path: string): string {
    if (
        typeof chrome !== "undefined" &&
        chrome.runtime &&
        chrome.runtime.getURL
    ) {
        return chrome.runtime.getURL(path);
    } else {
        return `/images/${path}`;
    }
}
