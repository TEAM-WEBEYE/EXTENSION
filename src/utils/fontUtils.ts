import type { FontSize, FontWeight } from "@src/contexts/ThemeContext";

export const fontSizeClassMap: Record<
    FontSize,
    { heading: string; common: string; caption: string }
> = {
    xl: {
        heading: "text-[32px]",
        common: "text-[28px]",
        caption: "text-[24px]",
    },
    l: {
        heading: "text-[30px]",
        common: "text-[26px]",
        caption: "text-[22px]",
    },
    m: {
        heading: "text-[28px]",
        common: "text-[24px]",
        caption: "text-[20px]",
    },
    s: {
        heading: "text-[26px]",
        common: "text-[22px]",
        caption: "text-[18px]",
    },
    xs: {
        heading: "text-[24px]",
        common: "text-[20px]",
        caption: "text-[16px]",
    },
};

export const fontWeightClassMap: Record<FontWeight, string> = {
    xbold: "font-extrabold",
    bold: "font-bold",
    regular: "font-normal",
};

export function getFontClasses(fontSize: FontSize, fontWeight: FontWeight) {
    const base = fontSizeClassMap[fontSize];
    const weight = fontWeightClassMap[fontWeight];
    return {
        fontHeading: `${base.heading} ${weight}`,
        fontCommon: `${base.common} ${weight}`,
        fontCaption: `${base.caption} ${weight}`,
    };
}
