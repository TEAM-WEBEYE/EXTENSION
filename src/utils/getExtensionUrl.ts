export function getExtensionUrl(path: string): string {
    if (
        typeof chrome !== "undefined" &&
        chrome.runtime &&
        chrome.runtime.getURL
    ) {
        return chrome.runtime.getURL(`/images/${path}`);
    } else {
        return `/images/${path}`;
    }
}
