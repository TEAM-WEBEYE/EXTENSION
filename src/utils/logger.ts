/* eslint-disable no-console */

type LogArgs = Array<
    string | number | boolean | object | null | undefined | unknown
>;

export const logger = {
    error: (message: string, ...args: LogArgs) => {
        console.error(`[ERROR] ${message}`, ...args);
    },

    warn: (message: string, ...args: LogArgs) => {
        console.warn(`[WARN] ${message}`, ...args);
    },

    debug: (message: string, ...args: LogArgs) => {
        if (process.env.NODE_ENV !== "production") {
            console.log(`[DEBUG] ${message}`, ...args);
        }
    },
};
