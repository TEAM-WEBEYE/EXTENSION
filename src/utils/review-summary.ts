export async function postReviewData(data: any) {
    try {
        const response = await fetch(
            "https://voim.store/api/v1/review/summary",
            {
                method: "POST",
                headers: {
                    "Content-Type": "application/json",
                },
                body: JSON.stringify(data),
            },
        );

        if (!response.ok) {
            throw new Error(`API 요청 실패: ${response.status}`);
        }

        console.log("✅ 서버에 리뷰 데이터 전송 완료");
    } catch (error) {
        console.error("❌ 리뷰 데이터 전송 실패:", error);
    }
}
