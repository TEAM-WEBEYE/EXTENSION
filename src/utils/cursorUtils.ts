import { getExtensionUrl } from "@src/utils/getExtensionUrl";
import type { CursorTheme, CursorSize } from "@src/contexts/ThemeContext";

export const cursorImages: Record<CursorTheme, Record<CursorSize, string>> = {
    white: {
        small: "cursors/white_small.png",
        medium: "cursors/white_medium.png",
        large: "cursors/white_large.png",
    },
    purple: {
        small: "cursors/purple_small.png",
        medium: "cursors/purple_medium.png",
        large: "cursors/purple_large.png",
    },
    yellow: {
        small: "cursors/yellow_small.png",
        medium: "cursors/yellow_medium.png",
        large: "cursors/yellow_large.png",
    },
    mint: {
        small: "cursors/mint_small.png",
        medium: "cursors/mint_medium.png",
        large: "cursors/mint_large.png",
    },
    pink: {
        small: "cursors/pink_small.png",
        medium: "cursors/pink_medium.png",
        large: "cursors/pink_large.png",
    },
    black: {
        small: "cursors/black_small.png",
        medium: "cursors/black_medium.png",
        large: "cursors/black_large.png",
    },
};

export function applyCursorStyle(
    theme: CursorTheme,
    size: CursorSize,
    enabled: boolean,
) {
    const styleId = "custom-cursor-style";
    const cursorUrl = getExtensionUrl(cursorImages[theme][size]);

    const styleContent = enabled
        ? `body, button, a, input, select, textarea { cursor: url('${cursorUrl}'), auto !important; }`
        : "";

    const updateStyleInDocument = (doc: Document) => {
        const existing = doc.getElementById(styleId);
        if (existing) existing.remove();

        if (enabled) {
            const style = doc.createElement("style");
            style.id = styleId;
            style.textContent = styleContent;
            doc.head.appendChild(style);
        }
    };

    updateStyleInDocument(document);
    document.querySelectorAll("iframe").forEach((iframe) => {
        try {
            const iframeDoc =
                iframe.contentDocument || iframe.contentWindow?.document;
            if (iframeDoc) updateStyleInDocument(iframeDoc);
        } catch (e) {
            console.warn("iframe 커서 적용 실패:", e);
        }
    });
}
