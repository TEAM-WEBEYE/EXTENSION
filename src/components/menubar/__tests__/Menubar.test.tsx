import "@testing-library/jest-dom";
import React from "react";
import { render, fireEvent, screen, act } from "@testing-library/react";
import Menubar from "../component";
import { logger } from "@src/utils/logger";

beforeEach(() => {
    global.chrome = {
        runtime: {
            sendMessage: jest
                .fn()
                .mockResolvedValue({ success: true }) as jest.Mock,
        },
    } as any;
});

afterEach(() => {
    jest.restoreAllMocks();
});

describe("Menubar", () => {
    it("컴포넌트가 올바르게 렌더링된다 (스냅샷)", () => {
        const { container } = render(
            <Menubar isOpen={true} onClose={() => {}}>
                hello
            </Menubar>,
        );

        expect(container).toMatchSnapshot();
    });

    it("isOpen이 true이면 모달이 보여야 한다", () => {
        render(
            <Menubar isOpen={true} onClose={() => {}}>
                hello
            </Menubar>,
        );
        expect(screen.getByText("설정 초기화")).toBeInTheDocument();
        expect(screen.getByAltText("나가기")).toBeInTheDocument();
    });

    it("isOpen이 false이면 모달이 안 보여야 한다", () => {
        render(
            <Menubar isOpen={false} onClose={() => {}}>
                hello
            </Menubar>,
        );
        expect(screen.queryByText("설정 초기화")).toBeNull();
        expect(screen.queryByAltText("나가기")).toBeNull();
    });

    it("Overlay 클릭 시 onClose가 호출된다", () => {
        const onClose = jest.fn();
        const { container } = render(
            <Menubar isOpen={true} onClose={onClose}>
                hello
            </Menubar>,
        );

        const overlay = container.querySelector("div");
        fireEvent.click(overlay!);

        expect(onClose).toHaveBeenCalled();
    });

    it("나가기 버튼 클릭 시 onClose가 호출된다", () => {
        const onClose = jest.fn();
        render(
            <Menubar isOpen={true} onClose={onClose}>
                hello
            </Menubar>,
        );

        const exitButton = screen.getByAltText("나가기");
        fireEvent.click(exitButton);

        expect(onClose).toHaveBeenCalled();
    });

    it("설정 초기화 버튼 클릭 시 chrome.runtime.sendMessage가 호출된다", async () => {
        render(
            <Menubar isOpen={true} onClose={() => {}}>
                hello
            </Menubar>,
        );

        const resetButton = screen.getByAltText("설정 초기화");
        fireEvent.click(resetButton);

        expect(chrome.runtime.sendMessage).toHaveBeenCalledWith({
            type: "RESET_SETTINGS",
        });
    });

    it("children이 제대로 렌더링된다", () => {
        render(
            <Menubar isOpen={true} onClose={() => {}}>
                <div>child</div>
            </Menubar>,
        );
        expect(screen.getByText("child")).toBeInTheDocument();
    });

    it("설정 초기화 버튼 클릭 시 chrome.runtime.sendMessage 실패 시 에러 로그가 찍힌다", async () => {
        const errorMessage = "Failed to send message";

        (chrome.runtime.sendMessage as jest.Mock).mockRejectedValueOnce(
            new Error(errorMessage),
        );

        const logError = jest
            .spyOn(logger, "error")
            .mockImplementation(() => {});

        render(
            <Menubar isOpen={true} onClose={() => {}}>
                hello
            </Menubar>,
        );

        const resetButton = screen.getByAltText("설정 초기화");

        await act(async () => {
            fireEvent.click(resetButton);

            await new Promise((resolve) => setTimeout(resolve, 0));
        });

        expect(logError).toHaveBeenCalledWith(
            "메시지 전송 중 오류:",
            expect.any(Error),
        );

        logError.mockRestore();
    });
});
