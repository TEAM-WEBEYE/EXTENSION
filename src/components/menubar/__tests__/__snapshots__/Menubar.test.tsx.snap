// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menubar 컴포넌트가 올바르게 렌더링된다 (스냅샷) 1`] = `
<div>
  <div
    class="css-17yo56f"
  >
    <div
      class="font-koddi bg-grayscale-100 css-6tuf1p"
    >
      <div
        class="flex justify-between mb-6 font-24-Bold"
      >
        <div
          class="px-6 py-[18px] flex gap-[10px] items-center cursor-pointer"
        >
          <img
            alt="설정 초기화"
            src="/images/arrow-rotate.png"
          />
          <div>
            설정 초기화
          </div>
        </div>
        <div
          class="py-[18px] cursor-pointer"
        >
          <img
            alt="나가기"
            src="/images/delete.png"
          />
        </div>
      </div>
      <div
        class="flex flex-col gap-5"
      >
        hello
      </div>
    </div>
  </div>
</div>
`;
